test/rules/prefer_const_constructors_in_immutables.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_const_constructors_in_immutables`
+
+// Hack to work around issues importing `meta.dart` in tests.
+// Ideally, remove:
+library meta;
```

Comment:
Does this deal correctly with factory constructors? redirecting constructors?
There aren't any cases where the constructor has a non-empty body, but there should be.